Userland/Libraries/LibWeb/Streams/ReadableStreamDefaultReader.cpp

```diff
@@ -206,8 +206,12 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<WebIDL::Promise>> ReadableStreamDefaultRead
 
     auto promise = WebIDL::create_promise(realm);
 
-    auto success_steps = [promise, &realm](ByteBuffer bytes) {
-        auto buffer = JS::ArrayBuffer::create(realm, move(bytes));
+    auto success_steps = [promise, &realm](Vector<ByteBuffer> const& byte_chunks) {
+        ByteBuffer concatenated_byte_chunks;
+        for (auto const& chunk : byte_chunks)
+            concatenated_byte_chunks.append(chunk.data(), chunk.size());
```

Comment:
I think we can just append the chunk directly:
```cpp
            concatenated_byte_chunks.append(chunk);
```

(and minor typo `BytteBuffer` in commit message)